### PR TITLE
ACA-1770: User with "Contributor" role on a site can't add a comment for a file/folder

### PR DESCRIPTION
**Pull request by amohammedalfresco**
- Fixed missing `Add` comment option for user who may not own a file but has privileges to contribute it
Alfresco docs: https://docs.alfresco.com/content-services/latest/using/permissions/#content-permissions

**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
In a repository service, a content not owned by a user but has been assigned with sufficient privileges to modify is not reflecting in ACA.  This behaviour is restricting adding a comment to a file that is not owned by a user of the site but been granted privileges (ex: `Contributor`).  The expected behaviour is available via RestApi and Share.

**What is the new behaviour?**
A proposed solution to remove extra layer of restriction as it is interfering with the original functionality of repository service.

**Does this PR introduce a breaking change?** (check one with "x")
Not sure :(
> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: src/app/components/info-drawer/comments-tab/comments-tab.component.ts [hunk 1]
```diff
@@ -25,26 +25,12 @@
...
-
-  constructor(private permission: NodePermissionService) {}
-
-  get canUpdateNode(): boolean {
-    if (!this.node) {
-      return false;
-    }
-
-    if (this.node.isFolder || (this.node.isFile && !isLocked({ entry: this.node }))) {
-      return this.permission.check(this.node, ['update']);
```
[the remark this answers is not in the document]
Comment (amohammedalfresco) [reply src/app/components/info-drawer/comments-tab/comments-tab.component.ts:46]: Thanks for the review and sorry for a very late update on this.  I just refactored the code to guard scenarios where the current user is not the owner of the node, but has a role (Contributor, Collaborator or Manager) which allow it to comment.

**Comment by amohammedalfresco** [comment]
I need to add unit tests to support changes/scenarios.

File: src/app/components/info-drawer/comments-tab/comments-tab.component.ts [hunk 2]
```diff
@@ -34,17 +37,37 @@ import { NodePermissionService, isLocked } from '@alfresco/aca-shared';
 export class CommentsTabComponent {
   @Input()
   node: MinimalNodeEntryEntity;
+  private userGroups: Group[] = [];
 
-  constructor(private permission: NodePermissionService) {}
+  // comment: Comment;
```
Comment (eromano) [inline src/app/components/info-drawer/comments-tab/comments-tab.component.ts:42]: maybe you forget this comment

File: projects/aca-shared/src/lib/services/node-permission.service.ts [hunk 3]
```diff
@@ -97,4 +97,35 @@ export class NodePermissionService implements NodePermissions {
...
+    return nodePermissions.inherited;
+  }
+
+  hasUserAuthorityOnNode(node: NodeEntry, userGroups: Group[]): boolean {
+    const nodeAuthorities: PermissionElement[] = this.getNodePermissions(node);
+
+    for (const nodeAuth of nodeAuthorities) {
+      for (const userAuth of userGroups) {
+        if (nodeAuth.authorityId === userAuth.id) {
+          if (nodeAuth.name === 'SiteContributor' || nodeAuth.name === 'SiteCollaborator' || nodeAuth.name === 'SiteManager') {
```
Comment (eromano) [inline projects/aca-shared/src/lib/services/node-permission.service.ts:123]: there is an enum in the JS-API for SiteContributor SiteCollaborator SiteManager better to use it

File: src/app/components/info-drawer/comments-tab/comments-tab.component.ts [hunk 4]
```diff
@@ -34,17 +37,37 @@ import { NodePermissionService, isLocked } from '@alfresco/aca-shared';
...
-      return this.permission.check(this.node, ['update']);
+      return (
+        this.nodePermissionService.check(this.node, ['update']) || this.nodePermissionService.hasUserAuthorityOnNode(<any>this.node, this.userGroups)
+      );
     }
     return false;
   }
+
+  getCurrentUserGroups() {
+    from(this.groupService.listAllGroupMembershipsForPerson('-me-', { maxItems: 250 }))
```
Comment (eromano) [inline src/app/components/info-drawer/comments-tab/comments-tab.component.ts:61]: what happen for more of 250?

**Review by eromano — CHANGES_REQUESTED** [review]

**Pull request closed without merging**